File: backend/schemas/product.py
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ProductBase(BaseModel):
    sku_code: str
    product_name: str
    price: Decimal
    quantity: int


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    sku_code: Optional[str] = None
    product_name: Optional[str] = None
    price: Optional[Decimal] = None
    quantity: Optional[int] = None


class ProductResponse(ProductBase):
    id: int
    model_config = ConfigDict(from_attributes=True)

File: frontend/types/product.ts
import type { ApiMessageResponse, DecimalString } from "./api";

interface ProductBase {
  sku_code: string;
  product_name: string;
  price: DecimalString;
  quantity: number;
}

export type ProductCreate = ProductBase;

export interface ProductUpdate {
  sku_code?: string;
  product_name?: string;
  price?: DecimalString;
  quantity?: number;
}

export interface ProductResponse extends ProductBase {
  id: number;
}

export type ProductListResponse = ProductResponse[];
export type ProductDeleteResponse = ApiMessageResponse;
